src/uct/api/uct_def.h

```diff
@@ -21,9 +22,23 @@
 #define UCT_TAG_PRIV_LEN         32
 #define UCT_AM_ID_BITS           5
 #define UCT_AM_ID_MAX            UCS_BIT(UCT_AM_ID_BITS)
-#define UCT_INVALID_MEM_HANDLE   NULL
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief  Initial values for memory registration keys.
+ *
+ * @{
+ */
+#define UCT_INVALID_MEM_HANDLE   NULL              /**< Initial value for the
+                                                        memory local key handle */
+#define UCT_LAZY_MEM_HANDLE      ((void *)(-1))    /**< The key handle for the memory
```

Comment:
I think this name implies "lazy registration" which is not the case, need a better name+explanation
need to document that data is copied to internal buffer, so UCP/user could estimate the cost, and guarantee the pointer can be dereferenced on CPU (e.g not GPU memory)
maybe this?
```
#define UCT_MEM_HANDLE_INVALID
#define UCT_MEM_HANDLE_DO_BCOPY
```